### ISSUE TITLE
Geofencing sample is still targeting sdk 28

### ISSUE THREAD

**Issue by BhavyaRattan** [googlesamples/android-play-location#200]
Geofencing sample is still targeting sdk 28: Will Geofences work in the same after targeting Android Q or are there any changes to be made?

**Comment by SyedAhmedJamil** [comment]
I don't think there is anything to change considering this basic Geo-fencing demo when targeting sdk version 29. Have you tried running by changing the sdk from 28 to 29 and running it on a Android Q device  or emulator ? Are you facing any issues ?

**Comment by BhavyaRattan** [comment]
Yes I tried running by changing sdk from 28 to 29 on an Android Q emulator, the issue there is that Geofencing requires BACKGROUND_LOCATION which is not currently handled in this repo. So you do not get any geofences triggered and the user doesn't even get to know why so. It should be handled in a similar way like the **locationupdatespendingintent** sample in this repo.

**Comment by SyedAhmedJamil** [comment]
Have you tried following the docs on  [Change location settings](https://developer.android.com/training/location/change-location-settings#java)

I don't think its sdk issue. This demo supresses missing permission warning using the  `@SuppressWarnings("MissingPermission")` assuming user will grant these permission manually either opening some other similar app like google maps and accepting location permission or going to app settings itself.

You have to implement location settings yourself for this demo. Check out the [MainActivity.java](https://github.com/googlesamples/android-play-location/blob/master/LocationUpdatesPendingIntent/app/src/main/java/com/google/android/gms/location/sample/locationupdatespendingintent/MainActivity.java) from _**locationupdatespendingintent**_ and implement the following functions either copy pasting or writing yourself

1. checkPermissions()
2. rquestPermissions()
3. onRequestPermissionsResult()

The check the `onCreate()` function and see how these function are called.

**Comment by pramodit** [comment]
Hey @BhavyaRattan  , It works with the JobScheduler API 

Refer this -  https://github.com/googlesamples/android-play-location/issues/197
